ui/desktop/app/controllers/scopes/scope/projects/targets/index.js

```diff
@@ -100,4 +108,37 @@ export default class ScopesScopeProjectsTargetsIndexController extends Controlle
       session_id
     );
   }
+
+  /**
+   * @param {object} event
+   */
+  @action
+  async handleInput(event) {
+    const { value } = event.target;
+    this.freeSearch = value;
+    await this.search();
+  }
+
+  /**
+   * builds a query string for all searchable target properties
+   * @returns {string}
+   */
+  buildQuery() {
+    return this.searchableProps
+      .map((prop) => `${prop} % '${this.freeSearch}'`)
+      .join(' or ');
+  }
```

Comment:
I think we talked about this before but does it make sense for this to live on the controller? Do you think the responsibility of building the query string should be on this component/controller?

Reply:
Moved out.